src/trigger-circleci-workflows.ts

```diff
@@ -106,7 +115,48 @@ async function runBot(context: probot.Context): Promise<void> {
   }
 }
 
+export function genHelpComment(config: object): string {
+  const availableLabels = Object.keys(config)
+    .map(label => {
+      return `* \`${label}\``;
+    })
+    .join('\n');
+  return `## :robot: This pull request has options for additional testing! :robot:
+
+Apply the following labels to run more tests for your pull request:
+${availableLabels}
+
+**NOTE**: These may be out of date, for a full list of options consult your repositories [configuration](../blob/master/.github/${configName})
+
+---
+Comment made with [pytorch-probot](https://github.com/pytorch/pytorch-probot)`;
+}
+async function postHelpComment(context: probot.Context): Promise<void> {
+  context.log.debug('Posting help comment!', 'postHelpComment');
+  if (isOnFork(context)) {
+    // Don't do anything
+    return;
+  }
+  const config = await loadConfig(context);
+  if (Object.keys(config).length === 0) {
+    context.log.debug(
+      `No configuration found for repository ${utils.repoKey(context)}`,
+      'trigger-circleci-workflows'
+    );
+    return;
+  }
+  const body = genHelpComment(config);
+  context.log.debug({body}, 'postHelpComment');
+  const createCommentResp = await context.github.issues.createComment(
+    context.issue({body})
+  );
+  context.log.info(
+    `Created usage comment on issue ${createCommentResp.data['html_url']}`
+  );
+}
+
 export function myBot(app: probot.Application): void {
+  app.on('pull_request.opened', postHelpComment);
```

Comment:
Yeah, definitely not convinced we want to spam every opened PR with this comment.  To give an example, rust-highfive (https://github.com/rust-lang/highfive) which is a similar bot that runs on every PR opened just comments to assign reviewers and welcome new contributors: if a PR is opened with reviewers already it will not say anything.

One possibility is we could have "@pytorchbot help" post the help comment, but only if you explicitly ask for it. But even better would just to have some wiki documentation for the feature, and maybe we can start programming a set of regexes for file paths of "is likely to need to run these extra jobs", where we can then post in a directed manner.

Reply:
Definitely have a wiki page. Preferably using GH's wiki.

Similar idea: track whose PRs we've shown the message to, and only show on the first PR. It'll still be a little spammy for maintainers, but less so. Either way, you probably only want to be aggressive about the messaging for this for some fixed time period.

Another idea: we add something to facehub indicating this feature exists. You won't capture the maintainers who aren't employees, nor employees on non-managed hardware/browsers, but it's like enough people most of the time.

Reply:
I'd probably say though with it being inside of a wiki that discoverability will be an issue and only the users who seek these kinds of things out will actually use it.

With putting this help message in every PR we're adding discoverability and people outside of the better engineering org will actually use it.

Reply:
> Similar idea: track whose PRs we've shown the message to, and only show on the first PR. It'll still be a little spammy for maintainers, but less so. Either way, you probably only want to be aggressive about the messaging for this for some fixed time period

This might work, but I don't think it's worth the added complexity of adding a DB to this.

Reply:
I agree that we should not add this comment to every PR; it creates a lot of noise for maintainers who get notified and keep track of lots of PRs/issues. Let's get some more experience with using the functionality before broadcasting it widely. Besides, me and Ed review a lot of diffs, so we can tell people it exists :P 

In the long run, what could be nice is for there to be a single comment on every PR that aggregates "build-related" information—which today is the Dr CI comment. It could detect which files you changed, and present something like:

> Based on files changed, you may want to run these additional CI jobs:
>    - rocm    # this is a link that actually runs the pipeline trigger

This would improve discoverability and relevancy, and would make it easier for users to trigger additional jobs. What do people think about this approach?